### DISCUSSION TITLE
Could a toggle be made so that Windhawk and VSCodium don't reach out online?

### DISCUSSION THREAD

**Discussion by TFWol** [ramensoftware/windhawk#91] [Ideas]
![image](https://github.com/ramensoftware/windhawk/assets/9045213/ab8c8c47-f229-44aa-9180-155ede5eab36)

Also could things like 'Fetches experiments to run from a Microsoft online service' be unchecked by default?
![image](https://github.com/ramensoftware/windhawk/assets/9045213/70d354f4-1778-46c1-a309-ee5c8ecf465b)

**Comment by m417z** [comment]
`mods.windhawk.net` is used for the mods catalog. Without it, the featured mods and the Explore view can't be displayed. I could introduce an offline mode, but I don't see a good reason to add it. If you really need it, you can use a firewall to block internet access to VSCodium.exe.

`raw.githubusercontent.com` can be used for mod images, for example it's used [here](https://windhawk.net/mods/shrink-address-bar-height).

`vscodium.now.sh` - I'd rather disable it, thanks for bringing it up. There are a couple of options, such as the "Fetch experiments" option you mentioned and extension update checking, that I'll disable in the next version of Windhawk.

**Reply by TFWol** [reply]
Yean, I tend to block internet by default; cloud everything is nerve wracking with all the breaches in the news. I understand your point though.

> vscodium.now.sh - I'd rather disable it, thanks for bringing it up. There are a couple of options, such as the "Fetch experiments" option you mentioned and extension update checking, that I'll disable in the next version of Windhawk.

No problem, I've recently started using the VSCodium IDE. I really like it, just had to trim out some telemetry. 
Can't find it all so I FW it and manually install downloaded extensions.

**Reply by eugenesvk** [reply]
> I could introduce an offline mode, but I don't see a good reason to add it. If you really need it, you can use a firewall to block internet access to VSCodium.exe.

Then you get errors in the main interface that obscure "Create a new mod" button

**Reply by m417z** [reply]
@eugenesvk Yes, but the error message can be closed.

**Reply by eugenesvk** [reply]
But it pops up again! It has to be closed every single time you visit the main page, which is a lot when you edit a mod (since recompiling doesn't pick up settings difference and you have to destr and create a new mod every time).

(And this popup also covers the confirm delete button.)

**Reply by m417z** [reply]
> since recompiling doesn't pick up settings difference and you have to destr and create a new mod every time

That's not how you're supposed to update settings. See my comment here: https://github.com/ramensoftware/windhawk/issues/440#issuecomment-2740296243.

> (And this popup also covers the confirm delete button.)

Only if the mod you're deleting is in a very specific position, which is extremely uncommon, unless you mean something else.

**Comment by m417z** [comment]
Done in version 1.4.

**Comment by dougbenham** [comment]
> ![image](https://user-images.githubusercontent.com/9045213/246943673-ab8c8c47-f229-44aa-9180-155ede5eab36.png)
> 
> Also could things like 'Fetches experiments to run from a Microsoft online service' be unchecked by default? ![image](https://user-images.githubusercontent.com/9045213/246943979-70d354f4-1778-46c1-a309-ee5c8ecf465b.png)

What app is doing that monitoring/blocking for you?

**Reply by TFWol** [reply]
That was from [DNSLookupView](https://www.nirsoft.net/utils/dns_lookup_view.html)
